ibm/service/eventstreams/data_source_ibm_event_streams_mirroring_config.go

```diff
@@ -0,0 +1,114 @@
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+
+}
+
+// read mirroring config using the admin-rest API
+func dataSourceIBMEventStreamsMirroringConfigRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	adminrestClient, err := meta.(conns.ClientSession).ESadminRestSession()
+	if err != nil {
+		tfErr := flex.TerraformErrorf(err, "Error getting Event Streams admin rest session", "ibm_event_streams_mirroring_config", "read")
+		log.Printf("[DEBUG]\n%s", tfErr.GetDebugMessage())
+		return tfErr.GetDiag()
+	}
+
+	adminURL, instanceCRN, err := getMirroringConfigInstanceURL(d, meta)
+	if err != nil {
+		tfErr := flex.TerraformErrorf(err, "Error getting Event Streams mirroring config URL", "ibm_event_streams_mirroring_config", "read")
+		log.Printf("[DEBUG]\n%s", tfErr.GetDebugMessage())
+		return tfErr.GetDiag()
+	}
+	adminrestClient.SetServiceURL(adminURL)
+
+	getMirroringConfigOptions := &adminrestv1.GetMirroringTopicSelectionOptions{}
+	mirroringConfig, _, err := adminrestClient.GetMirroringTopicSelectionWithContext(context, getMirroringConfigOptions)
+	if err != nil {
+		tfErr := flex.TerraformErrorf(err, "GetMirroringTopicSelection returned error", "ibm_event_streams_mirroring_config", "read")
+		log.Printf("[DEBUG]\n%s", tfErr.GetDebugMessage())
+		return tfErr.GetDiag()
+	}
+	if mirroringConfig == nil {
+		tfErr := flex.TerraformErrorf(err, "Unexpected nil config when getting mirroring config", "ibm_event_streams_mirroring_config", "read")
+		log.Printf("[DEBUG]\n%s", tfErr.GetDebugMessage())
+		return tfErr.GetDiag()
+	}
+	d.SetId(getMirroringConfigID(instanceCRN))
+	d.Set("resource_instance_id", instanceCRN)
+	d.Set("mirroring_topic_patterns", mirroringConfig.Includes)
+	return nil
+}
+
+func getMirroringConfigInstanceURL(d *schema.ResourceData, meta interface{}) (string, string, error) {
+	instanceCRN := d.Get("resource_instance_id").(string)
+	if instanceCRN == "" { // importing
+		id := d.Id()
+		crnSegments := strings.Split(id, ":")
+		if len(crnSegments) != 10 || crnSegments[8] != mirroringConfigResourceType {
+			return "", "", fmt.Errorf("ID '%s' is not a mirroring config resource", id)
+		}
+		crnSegments[8] = ""
+		crnSegments[9] = ""
+		instanceCRN = strings.Join(crnSegments, ":")
+		d.Set("resource_instance_id", instanceCRN)
+	}
+
+	instance, err := getInstanceDetails(instanceCRN, meta)
+	if err != nil {
+		return "", "", err
+	}
+	adminURL := instance.Extensions["kafka_http_url"].(string)
+	planID := *instance.ResourcePlanID
+	valid := strings.Contains(planID, "enterprise")
+	if !valid {
+		return "", "", fmt.Errorf("mirroring config is not supported by the Event Streams %s plan, enterprise plan is expected",
+			planID)
+	}
+	return adminURL, instanceCRN, nil
+}
+
+func getMirroringConfigID(instanceCRN string) string {
+	crnSegments := strings.Split(instanceCRN, ":")
+	crnSegments[8] = mirroringConfigResourceType
+	crnSegments[9] = ""
```

Comment:
In Event Streams, the first 8 segments identifies a service instance and the next two segments are for any resources on the instance. Segment 8 represents the resource type and segment 9 id of the resource.

Example mirroring-config id:

`ibm_event_streams_mirroring_config.es_mirroring_config crn:v1:bluemix:public:messagehub:us-south:a/6db1b0d0b5c54ee5c201552547febcd8:cb5a0252-8b8d-4390-b017-80b743d32839:mirroring-config:`

The type is mirroring-config. And since there can be one mirroring-config for a instance, it does not have a unique id. And we set segment 9 to "".